### PR TITLE
Integrate-Answers-in-QA-Page - Added Small Code

### LINKED ISSUE #885: [Frontend] Integrate Answers in Q&A Page
## Description
Currently the `Q&A` page only have the question list and form to submit the questions. There should be a way to see all the answers of the particular question sorted by the number of upvotes

##Expectation
Integrate the get answers api, filter the ones having `isApproved` as `true` and display it in nice way on the UI
Get Answers api - https://github.com/HITK-TECH-Community/Community-Website/blob/main/backend/app/routes/Q%26A/answers/getAnswers.js 

Screenshot 2024-05-15 at 8 57 48 PM

### PR DESCRIPTION
**Pull request by tejasnasre**
Closes: #885 (Draft PR)

## Proposed changes
Currently the Q&A page only have the question list and form to submit the questions. There should be a way to see all the answers of the particular question sorted by the number of upvotes

### Brief description of what is fixed or changed
Added The Answer Page For Specific Question When  User Clicks On The Get Answer Button

## Types of changes

- [x] Integrating Front End Answer For Specific Question. (Non -Breaking Changes)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots
Added Button In Question Card :
Answer Added

Answer Page For Specific Question By User IDs : 
answerpage

## Other information

There Answer In Past Days But They Deleted The Answers And Question Also So I Will Work On The Issue  #886 And Then Back To This 😁

### CODE REVIEW

File: backend/.env [hunk 1]
```diff
@@ -1,4 +1,4 @@
-MONGO_DB_URL=<mongoDB_URL>
+MONGO_DB_URL=mongodb://localhost:27017/testdb
```
Comment (Kajol-Kumari) [inline backend/.env:1]: don't push .env changes

File: frontend/src/app.js [hunk 2]
```diff
@@ -132,6 +133,11 @@ const App = () => {
                 path="/Q&A"
                 render={() => <Ques theme={theme} />}
               />
+              <Route
```
Comment (Kajol-Kumari) [inline frontend/src/app.js:136]: why do we need a separate route for answers?? each question should have a list of answers right?

File: frontend/src/pages/Q&A/Q&A.jsx [hunk 3]
```diff
@@ -273,6 +269,9 @@ function Ques(props) {
                   >
                     👎 {item.downvote}
                   </button>
+                  <button>
+                    <Link to={`/getanswers/${item._id}`}>Get Answer</Link>
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Q&A/Q&A.jsx:273]: instead to directing to a different page, make it like a pop-up, (check `Broadcast` page for reference)

File: frontend/src/pages/Q&A/Components/GetAnswer/GetAnswer.jsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        });
+    } catch (error) {
+      console.error("Error fetching answer:", error);
+    }
+  };
+  useEffect(() => {
+    GetAnswer(answerId);
+  }, []);
+
+  console.log(answer);
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Q&A/Components/GetAnswer/GetAnswer.jsx:36]: please remove console log statement.

File: frontend/src/pages/Q&A/Components/GetAnswer/GetAnswer.jsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+function GetAnswer() {
+  const { answerId } = useParams();
+  // console.log(answerId);
+
+  const [answer, setAnswer] = useState([]);
+
+  //Get Answer
+  const GetAnswer = (answerId) => {
+    try {
+      fetch(`${END_POINT}/answers/${answerId}`, {
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Q&A/Components/GetAnswer/GetAnswer.jsx:17]: Add page loader and generic toast message Which should show success message on success and failure message on failure.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]

**Comment by TejasNasre** [comment]
I will add all this soon I have exam that why I am inactive it's draft PR

File: frontend/src/app.js (hunk 2 above)
Comment (TejasNasre) [reply frontend/src/app.js:136]: I think it's more  Efficient for  Data Handling

Comment (Kajol-Kumari) [reply frontend/src/app.js:136]: directing to a new page for answers doesn't make sense, either make it collapsible(like FAQ page) or open a pop-up and show the question and all its answers,

**Pull request closed without merging**